lib/watcher_adapter.js

```diff
@@ -27,12 +28,13 @@ module.exports = class WatcherAdapter extends EventEmitter {
     if (!Array.isArray(watchedPaths)) {
       throw new TypeError(`WatcherAdapter#watch's first argument must be an array of watchedPaths`);
     }
-    let watchers = watchedPaths.map(watchedPath => {
+    let watchers = watchedPaths.map(({ root: watchedRoot, path: watchedPath }) => {
       const watcher = new sane(watchedPath, this.options);
       this.watchers.push(watcher);
-      bindFileEvent(this, watcher, 'change');
-      bindFileEvent(this, watcher, 'add');
-      bindFileEvent(this, watcher, 'delete');
+      bindFileEvent(this, watcher, watchedRoot, 'change');
+      bindFileEvent(this, watcher, watchedRoot, 'add');
+      bindFileEvent(this, watcher, watchedRoot, 'delete');
```

Comment:
Instead of this, do `this.on('change', () => watchedRoot.revise())`

Reply:
The issue is that I would have to do that for change, add, and delete which triples the places where we would have to call revise. Doing it this way keeps the change diff down and centralizes the ownership of the `watchedRoot.revise()` call.

Reply:
I don’t understand. Change is triggered whenever a file is added/changed/removed. You’d only need one event handler.

Reply:
Adding:

```js
this.on('change', () => {
  console.log('changed fired');
  watchedRoot.revise();
});
```

This callback gets fired many times even for a single change (29 times per file in the case of travis-web). To fix this I could do 

```js
watcher.on('change', () => {
  console.log('changed fired');
  watchedRoot.revise();
});
```

Which will only fire once per change but it does not pick up adding new files or deleting new files. I believe we are going to have to keep the logic the way it is in this PR so that we don't run into these problems.